### PR TITLE
Chore/set up the app mobile architecture

### PR DESCRIPTION
**Pull request by MaximeHavez**
# Description
Add Flutter librairies (riverpod, easy localization, retrofit, json-serializable, go-router)

# Linked Issues
#14 

# Changes
New tools and documentations

# Screenshots
N/A

# Tests
How has it been tested ?

- [x] Manually
- [ ] Automated tests
- [ ] QA
- [ ] Other

# Additional information
N/A

### CODE REVIEW

File: yaki_mobile/pubspec.yaml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  # can now use : import 'package:go_router/go_router.dart';
+  go_router: ^6.0.7
+  flutter:
+    sdk: flutter
+
+  # // Riverpod dependency //
+  # This dependency allows for dependency injection and manage states
+  # import 'package:hooks_riverpod/hooks_riverpod.dart'
+  # link : https://riverpod.dev/docs/getting_started
+  flutter_hooks: ^0.18.0
```
Comment (PiotrFLEURY) [inline yaki_mobile/pubspec.yaml:51]: Remove

File: yaki_mobile/pubspec.yaml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  hooks_riverpod: ^2.2.0
```
Comment (PiotrFLEURY) [inline yaki_mobile/pubspec.yaml:52]: Remove

File: yaki_mobile/pubspec.yaml [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
```
Comment (PiotrFLEURY) [inline yaki_mobile/pubspec.yaml:53]: Add flutter_riverpod see [https://pub.dev/packages/flutter_riverpod/install](https://pub.dev/packages/flutter_riverpod/install)

File: yaki_mobile/pubspec.yaml [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+  # easy_localization : manage app translation, use folder defined in assets "translations", set .json (or others files) in this folder
+  # https://pub.dev/packages/easy_localization
+  # can now use : import 'package:json_serializable/json_serializable.dart';
+  easy_localization: 3.0.1
+  # routing package for Flutter that uses the Router API to provide a convenient,
+  # url-based API for navigating between different screens
+  # https://pub.dev/packages/go_router
+  # can now use : import 'package:go_router/go_router.dart';
+  go_router: ^6.0.7
+  flutter:
```
Comment (PiotrFLEURY) [inline yaki_mobile/pubspec.yaml:44]: sort dependencies

File: yaki_mobile/pubspec.yaml [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+# To automatically upgrade your package dependencies to the latest versions
+# consider running `flutter pub upgrade --major-versions`. Alternatively,
+# dependencies can be manually updated by changing the version numbers below to
+# the latest version available on pub.dev. To see which dependencies have newer
+# versions available, run `flutter pub outdated`.
+
+dependencies:
+  # json_serializable : Provide a Dart build system to handle JSON
+  # https://pub.dev/packages/json_serializable
+  json_serializable: ^6.6.1
```
Comment (PiotrFLEURY) [inline yaki_mobile/pubspec.yaml:34]: Remove

**Review by PiotrFLEURY — CHANGES_REQUESTED** [review]
Please fix pubspec.yaml

File: yaki_mobile/pubspec.yaml [hunk 6]
```diff
@@ -0,0 +1,118 @@
+name: yaki
+description: A new Flutter project.
```
Comment (PiotrFLEURY) [inline yaki_mobile/pubspec.yaml:2]: set a clear description of the app

File: yaki_mobile/pubspec.yaml [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+  sdk: ">=2.19.2 <3.0.0"
+
+# Dependencies specify other packages that your package needs in order to work.
+# To automatically upgrade your package dependencies to the latest versions
+# consider running `flutter pub upgrade --major-versions`. Alternatively,
+# dependencies can be manually updated by changing the version numbers below to
+# the latest version available on pub.dev. To see which dependencies have newer
+# versions available, run `flutter pub outdated`.
+
+dependencies:
```
Comment (PiotrFLEURY) [inline yaki_mobile/pubspec.yaml:31]: some dependencies are still missing. Please read the linked Google doc in the issue.

**Review by PiotrFLEURY — CHANGES_REQUESTED** [review]
The create base folders for Flutter front architecture pattern are missing. Please check issue

**Review by PiotrFLEURY — APPROVED** [review]

**Review by FlorianFlouquet — APPROVED** [review]

**Pull request merged**